### ISSUE TITLE
esp收集到的原始数据就是整数吗？ (AEGHB-421)

### ISSUE THREAD

**Issue by non-1024** [espressif/esp-csi#139]
这是我是用esp-csi这个项目收集到的数据：
![image](https://github.com/espressif/esp-csi/assets/48765322/77c4252b-260c-420b-9f8f-3b790b2b5787)

![G$~R{NS18WYM$U9LZPSANQB](https://github.com/espressif/esp-csi/assets/48765322/343c4e3a-9bdb-4059-9e9f-aae9b17e30de)

我想请问一下，esp收集到的原始数据都是整数吗？数据是否经过处理之后才保存？
因为我看其他人使用intel-5300收集到的数据是有小数点的，不会像esp这样这么整整齐齐。
谢谢！

**Comment by Yanc14** [comment]
> 这是我用 esp-csi 这个项目收集到的数据： ![图像](https://user-images.githubusercontent.com/48765322/273891852-77c4252b-260c-420b-9f8f-3b790b2b5787.png)
> 
> ![G$~R{NS18WYM$U9LZPSANQB](https://user-images.githubusercontent.com/48765322/273894027-343c4e3a-9bdb-4059-9e9f-aae9b17e30de.png)
> 
> 我想问一下，esp收集到的原始数据都是整数吗？数据是否经过处理之后才保存？ 因为我看其他人使用intel-5300收集到的数据是有小数据点的，不会像esp这样这么整啊齐。 谢谢！

你好！请问你是路由器+esp，还是双esp的机制呢？

**Comment by Yanc14** [comment]
> 路由器发射，esp32接收

你这个数据接收是怎存储数据的呀？TF卡还是串口发送到上位机存储？

**Comment by NorthOfTheNorthDistrict** [comment]
> > 路由器发射，esp32接收
> 
> 你这个数据接收是怎存储数据的呀？TF卡还是串口发送到上位机存储？

不好意思，我本人并不懂硬件。我使用这个esp-csi的开源项目采集数据，至于是怎样保存数据的我并不知道。我只是按照REANME文件烧录好程序之后，采集数据时会自动保存在我的笔记本电脑里面。

**Comment by MacChu0315-Espressif** [comment]
> 我想请问一下，esp收集到的原始数据都是整数吗？数据是否经过处理之后才保存？ 因为我看其他人使用intel-5300收集到的数据是有小数点的，不会像esp这样这么整整齐齐。 谢谢！

Yes, we do not perform any additional data processing. The principle of our CSI data acquisition is consistent with the 5300, except that we further provide 8-bit data (data range expanded to -128~127).

Furthermore, based on our testing, CSI data obtained through the widely accepted 5300 driver ([Linux 802.11n CSI Tool](https://dhalperi.github.io/linux-80211n-csitool/index.html#publications)) also appears as integer data like the following image:
![image](https://github.com/espressif/esp-csi/assets/130628192/e2de1303-5306-4587-bd74-22b739999567)

Please feel free to provide more information, so we can offer further assistance.
